pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -100,118 +134,204 @@ func unbindPCIDeviceFromVfioPCIDriver(addr string) error {
+		pdcCopy := pdc.DeepCopy()
+
+		// Try to unbind from existing driver, if applicable
+		err = unbindDeviceFromDriver(pd.Status.Address, pd.Status.KernelDriverInUse)
+		if err != nil {
+			pdcCopy.Status.PassthroughEnabled = true
+			logrus.Errorf("Error unbinding device after reboot: %s", err)
+		} else {
+			pdcCopy.Status.PassthroughEnabled = false
+		}
+
+		// Enable Passthrough on the device
+		err = enablePassthrough(pd)
+		if err != nil {
+			logrus.Errorf("Error rebinding device after reboot: %s", err)
+			pdcCopy.Status.PassthroughEnabled = false
+		} else {
+			pdcCopy.Status.PassthroughEnabled = true
+		}
+		_, err = h.pdcClient.UpdateStatus(pdcCopy)
+		if err != nil {
+			logrus.Errorf("Failed to update PCIDeviceClaim status for %s: %s", pdc.Name, err)
+		}
+	}
+}
+
+func (h Handler) reconcilePCIDeviceClaims(nodename string) error {
+	// Get all PCI Device Claims on this node
+	pdcs, err := h.pdcClient.List(metav1.ListOptions{})
+	if err != nil {
+		return err
 	}
 
 	// Only load the vfio drivers if there are any PCI Device Claims
 	if len(pdcs.Items) > 0 {
 		loadVfioDrivers()
 	}
 
-	// Get those PCI Device Claims for this node
 	for _, pdc := range pdcs.Items {
-		if pdc.Spec.NodeName == nodename {
-			if pdc.DeletionTimestamp != nil {
-				logrus.Infof("Attempting to unbind PCI device %s from vfio-pci", pdc.Spec.Address)
-				err = unbindPCIDeviceFromVfioPCIDriver(pdc.Spec.Address)
-				if err != nil {
-					return err
-				}
-			}
-			if !pdc.Status.PassthroughEnabled {
-				logrus.Infof("Attempting to enable passthrough")
-				// Get PCIDevice for the PCIDeviceClaim
-				name := pdNames[pdc.Spec.NodeAddr()]
-
-				pd, err := h.pdClient.Get(name, metav1.GetOptions{})
-				if err != nil {
-					return err
-				}
-				pdc.Status.KernelDriverToUnbind = pd.Status.KernelDriverInUse
-				if pd.Status.KernelDriverInUse == "vfio-pci" {
-					pdc.Status.PassthroughEnabled = true
-				} else {
-					// Only unbind from driver is a driver is currently in use
-					if strings.TrimSpace(pd.Status.KernelDriverInUse) != "" {
-						err = unbindPCIDeviceFromDriver(pd.Status.Address, pd.Status.KernelDriverInUse)
-						if err != nil {
-							pdc.Status.PassthroughEnabled = false
-							return err
-						}
-					}
-					err = addNewIdToVfioPCIDriver(pd.Status.VendorId, pd.Status.DeviceId)
-					if err != nil {
-						pdc.Status.PassthroughEnabled = false
-						return err
-					}
-					pdc.Status.PassthroughEnabled = true
-				}
-				_, err = h.pdcClient.UpdateStatus(&pdc)
-				if err != nil {
-					return err
-				}
-			}
+		// TODO change UI to label each PDC to allow for the LabelSelector instead of testing all PDCs
+		if pdc.Spec.NodeName != nodename {
+			continue
+		}
+		if !pdc.Status.PassthroughEnabled {
+			// 2022-09-26: 3:48PM PDT Removed the go because only one PDC was being created
+			h.attemptToEnablePassthrough(&pdc)
 		}
 	}
 
 	return nil
 }
+
+func (h Handler) attemptToEnablePassthrough(pdc *v1beta1.PCIDeviceClaim) {
```

Comment:
There are many error cases in this function, it should return error, other than swallow error